.github/actions/k-bench/evaluate/graph.py

```diff
@@ -0,0 +1,189 @@
+    'depl_scale':   'Deployment Scale',
+    'depl_delete':  'Deployment Delete',
+}
+
+fio_suffix = 'MiB/s'
+fio_key2header = {
+    'fio_root_async_R70W30_R':   'async_R70W30 mix,\n seq. reads',
+    'fio_root_async_R70W30_W':   'async_R70W30 mix,\n seq. writes',
+    'fio_root_async_R100W0_R':   'async_R100W0 mix,\n seq. reads',
+    'fio_root_async_R0W100_W':   'async_R0W100 mix,\n seq. writes',
+}
+
+net_suffix = 'Mbit/s'
+net_key2header = {
+    'net_internode_snd':            f'iperf internode \n send ({net_suffix})',
+    'net_intranode_snd':            f'iperf intranode \n send ({net_suffix})',
+}
+
+
+def configure() -> str:
+    """Read the benchmark data paths.
+
+    Expects ENV vars (required):
+    - BDIR=benchmarks
+
+    Raises TypeError if at least one of them is missing.
+
+    Returns: out_dir
+    """
+    out_dir = os.environ.get('BDIR', None)
+    if not out_dir:
+        raise TypeError(
+            'ENV variables BDIR is required.')
+    return out_dir
+
+
+def bar_chart(data, headers, title='', suffix='', val_label=True, y_log=False):
+    """Draws a bar chart with multiple bars per data point.
+
+    Args:
+        data (dict[str, list]): Benchmark data dictionary: subject -> lists of value points
+        headers (list): List of headers (x-axis).
+        title (str, optional): The title for the chart. Defaults to "".
+        suffix (str, optional): The suffix for values e.g. "MiB/s". Defaults to "".
+        val_label (bool, optional): Put a label of the value over the bar chart. Defaults to True.
+        y_log (bool, optional): Set the y-axis to a logarithmic scale. Defaults to False.
+    Returns:
+        fig (matplotlib.pyplot.figure): The pyplot figure
+    """
+    fig, ax = plt.subplots(figsize=(10, 5))
+    fig.patch.set_facecolor('white')
+
+    # Number of bars per group
+    n_bars = len(data)
+
+    # The width of a single bar
+    bar_width = 0.8 / n_bars
+
+    # List containing handles for the drawn bars, used for the legend
+    bars = []
+
+    # Iterate over all data
+    for i, values in enumerate(data.values()):
+        # The offset in x direction of that bar
+        x_offset = (i - n_bars / 2) * bar_width + bar_width / 2
+
+        # Draw a bar for every value of that type
+        for x, y in enumerate(values):
+            bar = ax.bar(x + x_offset, y, width=bar_width * 0.9,
+                         color=BAR_COLORS[i % len(BAR_COLORS)], edgecolor="black")
+            if val_label:
+                ax.bar_label(bar, fmt='%g {suffix}'.format(suffix=suffix))
+        # Add a handle to the last drawn bar, which we'll need for the legend
+        bars.append(bar[0])
+    # Draw legend
+    ax.legend(bars, LEGEND_NAMES)
+    if y_log:
+        ax.set_yscale('log')
+    ax.set_xticks(np.arange(len(headers)))
+    ax.set_xticklabels(headers)
+
+    plt.setp(ax.get_xticklabels(), fontsize=LABEL_FONTSIZE,
+             rotation=LABEL_ROTATE_BY)
+    plt.title(f'{title} ({suffix})')
```

Comment:
You're mixing use of `.format` and f-strings here. I think it would be nice to stick to one. I personally prefer f-strings, but chose as you like :)

Reply:
I prefer f strings as well, but I recently read the https://github.com/wemake-services/wemake-python-styleguide and tried to adapt a bit of it ;)